### PR TITLE
feat(util): add themeResolver

### PR DESCRIPTION
**Pull request by briantstephan**
This adds a themeResolver util that converts a ThemeConfig to a TailwindConfig format, and combines it with the developer-specified tailwind config that is also passed in. Configurations made by the developer take priority over the user/marketer's theme config. Tested by building locally and using the local build in the starter. I put part of the tailwind config into theme.config.ts and passed both the theme config and the rest of the tailwind config into themeResolver, which provided the same output as the old tailwind config.

### CODE REVIEW

File: src/utils/themeResolver.ts [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+const isObject = (value: any): value is PlainObject => {
+  return value !== null && typeof value === "object";
+};
+
+/**
+ * Merges the developer-specified theming and the marketer-specified theming,
+ * giving priority to the developer-specified theming.
+ */
+export const themeResolver = <T extends PlainObject>(
```
Comment (mkilpatrick) [inline src/utils/themeResolver.ts:94]: Add tests for these different functions

**Comment by mkilpatrick** [comment]
> add docs to READ.me for this too!

Idk if we need a README if it's internal

**Comment by asanehisa** [comment]
> > add docs to READ.me for this too!
> 
> Idk if we need a README if it's internal

Oh if its internal, it should be in src/internal/utils?

File: src/utils/themeResolver.ts [hunk 2]
```diff
@@ -91,10 +91,13 @@ const isObject = (value: any): value is PlainObject => {
...
  */
-export const themeResolver = <T extends PlainObject>(
-  deverloperTheming: T,
+export const themeResolver = (
+  deverloperTheming: TailwindConfig,
   marketerTheming: ThemeConfig
 ) => {
   const marketerTailwindConfig = convertToTailwindConfig(marketerTheming);
-  return deepMerge(deverloperTheming, marketerTailwindConfig);
+  console.log("marketerTailwindConfig", marketerTailwindConfig);
```
Comment (mkilpatrick) [inline src/utils/themeResolver.ts:99]: rm logs

Comment (briantstephan) [reply src/utils/themeResolver.ts]: Oops, done

**Review by benlife5 — APPROVED** [review]

File: src/utils/themeResolver.test.ts [hunk 3]
```diff
@@ -0,0 +1,322 @@
...
+    },
+  },
+  borderRadius: {
+    lg: "var(--radius)",
+    md: "calc(var(--radius) - 2px)",
+    sm: "calc(var(--radius) - 4px)",
+  },
+};
+
+const developerTailwindConfig: TailwindConfig = {
```
Comment (jwartofsky-yext) [inline src/utils/themeResolver.test.ts:240]: same

File: src/utils/themeResolver.test.ts [hunk 4]
```diff
@@ -0,0 +1,322 @@
...
+        label: "Small Border Radius",
+        plugin: "borderRadius",
+        type: "number",
+        default: "calc(var(--radius) - 4px)",
+      },
+    },
+  },
+};
+
+const marketerTailwindConfig: TailwindConfig = {
```
Comment (jwartofsky-yext) [inline src/utils/themeResolver.test.ts:200]: same

File: src/utils/themeResolver.test.ts [hunk 5]
```diff
@@ -0,0 +1,322 @@
...
+    "restructures theme config and merges with developer-specified tailwind config, " +
+      "prioritizing developer sepecifications",
+    () => {
+      const result = themeResolver(developerTailwindConfig, themeConfig);
+      expect(result).toEqual(mergedConfig);
+    }
+  );
+});
+
+const themeConfig: ThemeConfig = {
```
Comment (jwartofsky-yext) [inline src/utils/themeResolver.test.ts:31]: put this above the tests

**Review by jwartofsky-yext — CHANGES_REQUESTED** [review]
lgtm other than nitpicks

**Review by jwartofsky-yext — APPROVED** [review]
lgtm

**Review by mkilpatrick — APPROVED** [review]

File: src/utils/themeResolver.test.ts (hunk 4 above)
Comment (briantstephan) [reply src/utils/themeResolver.test.ts:200]: Ack

File: src/utils/themeResolver.test.ts (hunk 3 above)
Comment (briantstephan) [reply src/utils/themeResolver.test.ts:240]: Ack

**Pull request merged**